src/Gui/Application.cpp

```diff
@@ -2180,6 +2180,21 @@ void Application::runApplication()
 
             int major = context.format().majorVersion();
             int minor = context.format().minorVersion();
+
+#ifdef NDEBUG
+            // In release mode, issue a warning to users that their version of OpenGL is
+            // very old.
+            if (major < 2) {
+                auto message =
+                    QObject::tr("This system is running OpenGL %1.%2. "
```

Comment:
Is it really what is wanted to use 'QObject::tr', which means text will have no context for translation?

Reply:
That's what we're using elsewhere in this file. The class this is in isn't itself derived from `QObject`.

Reply:
Yes I saw that. So generally (when using Qt) either 'QCoreApplication::translate' is used, or class is equipped with the tr function by adding to it the 'Q_DECLARE_TR_FUNCTIONS' macro. So translations will have a context.
This said if at other places in the file there are other translations using direct call to 'QObject::tr' better use it too. But maybe later we have to think about contextualizing all translations.